### DISCUSSION TITLE
how to disable _interopRequireDefault in "@babel/preset-env" with "modules": "amd"?

### DISCUSSION THREAD

**Discussion by gabrielgortabns** [babel/babel#15185] [Q&A] [answered]
how to disable _interopRequireDefault in "@babel/preset-env" with "modules": "amd"?

I have ES modules, I need to transpile to AMD ones.

I have code:

```js
import something from 'something';
something()
```

and it transpiles to:

```js
define(["something"], function (_something) {
  "use strict";

  _something = _interopRequireDefault(_something);
  function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }
  (0, _something.default)();
});
```

I would like it to transpile to:

```js
define(["something"], function (_something) {
  "use strict";

  _something()
});
```

**Comment by gabrielgortabns** [comment]
ok, it seems it's impossible to do via configuration, so the only solution I found is to replace it "manually" via search-and-replace plugin, that will find `.default` string and replace it with empty string, and similiar, replace all lines containing `__interopRequireDefault` with empty string (resp. remove those lines).

**Reply by gabrielgortabns** [reply]
This approach was wrong too. Wrote plugin myself, simple visitor object that handles it.

**Comment by nicolo-ribaudo** [comment]
You can pass `modules: false`, and enable `@babel/plugin-transform-modules-amd` with the [`importInterop` option](https://babeljs.io/docs/en/babel-plugin-transform-modules-commonjs#importinterop).

**Reply by gabrielgortabns** [reply, marked as answer]
I see, I just solved it with writing my own babel plugin, that removes those functions 😄 will try your solutions (that I think will give better perf)

**Reply by gabrielgortabns** [reply]
I tried your approach, but it didn't worked. Instead I used preset-env to convert it to CJS, then I use plugins I wrote myself (implemented visitor object) to remove all `__interopRequireDefault`, `.default` and so on. And then I inject code to AMD file.
